### PR TITLE
[KED-2165] Remove get_project_context()

### PR DESCRIPTION
**Pull request by lorenabalan**
## Description

`get_project_context` was kept for compatibility with Kedro 0.14.*.
Now that we no longer support old versions of Kedro (<0.15), we can replace the calls with `load_context()` or `KedroSession.load_context()`. `get_project_context` will also be removed in future major release of Kedro.

## Development notes

- Dropped support for Kedro 0.15.x. 
- Removed legacy code handling dataset layers (`DataCatalog.layers` is a definite thing from 0.16.x onwards, no need to check for the old way of having the `layer` attribute on the dataset class).
- Removed legacy tests.
- Ignored covered on branch testing 0.17.0 workflow because that version hasn't been released yet. I could probably test with mocking a lot about but it feels a bit redundant given we'll have to remove that once we release. - Tested this branch manually instead, installing kedro from `develop`, and changing the `__version__` to 0.17.0 (as well as `project_version` in `pyproject.toml` in a newly generated kedro project).

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Review by limdauto — APPROVED** [review]
LGTM!

**Review by DmitriiDeriabinQB — APPROVED** [review]
LGTM! 🧑‍🍳

File: package/kedro_viz/server.py [hunk 1]
```diff
@@ -290,20 +279,12 @@ def find_child_layers(node_id: str) -> Set[str]:
...
-        dataset_to_layer = {
-            ds_name: getattr(ds_obj, "_layer", None)
-            for ds_name, ds_obj in _CATALOG._data_sets.items()
-        }
+    if _CATALOG.layers is None:
+        return {ds_name: None for ds_name in getattr(_CATALOG, "_data_sets")}
+
+    dataset_to_layer = {}
+    for layer, dataset_names in _CATALOG.layers.items():
+        dataset_to_layer.update({dataset_name: layer for dataset_name in dataset_names})
```
[the remark this answers is not in the document]
Comment (lorenabalan) [reply package/kedro_viz/server.py:287]: I find double for-loops in list comprehensions quite hard to read so I prefer to leave it as it was before.

File: package/tests/test_server.py [hunk 2]
```diff
@@ -188,7 +180,7 @@ def client():
     return client
 
 
-@pytest.mark.usefixtures("patched_get_project_context")
+@pytest.mark.usefixtures("patched_load_context")
```
[the remark this answers is not in the document]
Comment (lorenabalan) [reply package/tests/test_server.py]: Done. :)

**Pull request merged**